keylime/web_util.py

```diff
@@ -11,117 +10,195 @@
 import tornado.web
 
 from keylime import api_version as keylime_api_version
-from keylime import ca_util, config, json
+from keylime import ca_util, config, json, secure_mount
 
 
-def init_mtls(section="cloud_verifier", generatedir="cv_ca", logger=None):
+def get_tls_dir(component):
+    # Get the values from the configuration file
+    tls_dir = config.get(component, "tls_dir")
+
+    if not tls_dir:
+        raise Exception(f"The 'tls_dir' option is not set for '{component}'")
+
+    if tls_dir == "generate":
+        if component == "verifier":
+            generatedir = "cv_ca"
+        elif component == "registrar":
+            generatedir = "reg_ca"
+        else:
+            raise Exception(f"The tls_dir=generate option is not supported for " f"'{component}'")
+
+        tls_dir = os.path.abspath(os.path.join(config.WORK_DIR, generatedir))
+    elif tls_dir == "default":
+        if component in ("verifier", "registrar", "tenant"):
+            # Use the keys/certificates generated for the verifier
+            tls_dir = os.path.abspath(os.path.join(config.WORK_DIR, "cv_ca"))
+        elif component == "agent":
+            # For the agent, use the secure mount dir as the default directory
+            tls_dir = secure_mount.get_secdir()
+    else:
+        # if it is relative path, convert to absolute in WORK_DIR
+        if tls_dir[0] != "/":
```

Comment:
Not needed?
```
>>> import os.path
>>> WORK_DIR="/tmp"
>>> p1="/mydir"
>>> p2="mydir"
>>> os.path.join(WORK_DIR, p1)
'/mydir'
>>> os.path.join(WORK_DIR, p2)
'/tmp/mydir'
```

Reply:
Interesting, I didn't know this was the behavior.

Reference: https://docs.python.org/3/library/os.path.html#os.path.join